multimedia/VLC/Portfile

```diff
@@ -40,434 +30,673 @@ subport             lib${name} {
+
+    # VLC doesn't currently build for me using port:clang-3.8
+    compiler.blacklist-append \
+                        gcc-4.2 llvm-gcc-4.2 macports-clang-3.8 {clang < 300}
+
+    patchfiles          buildfix-package.mak.patch \
+                        configure.ac-no-arch.patch \
+                        PR-34741-no__clang_version__.patch \
+                        static_assert.patch \
+                        no-sparkle.patch \
+                        patch-vlc-no-O4.diff \
+                        patch-ffmpegVLC.diff
 
-compiler.blacklist  gcc-4.2 llvm-gcc-4.2 {clang < 300}
+    post-patch {
+        reinplace "s:librsvg-2/librsvg:librsvg:" \
+            ${worksrcpath}/modules/text_renderer/svg.c
 
-pre-fetch {
-    if {${os.platform} eq "darwin" && ${os.major} < 10} {
-        ui_error "${name} ${version} requires Mac OS X 10.6 or greater."
-        return -code error "incompatible Mac OS X version"
+        if {[string match *clang* ${configure.cxx}] && ${configure.cxx_stdlib} == "libc++"} {
+            reinplace "s:-lstdc\+\+:-lc++:" \
+                ${worksrcpath}/configure.ac \
+                ${worksrcpath}/modules/access/Makefile.am
+        }
+        # patch source using ffmpeg to use the correct header file locations
+        foreach dir {libavcodec libavformat libavresample libavutil libpostproc libswresample libswscale} {
+            foreach module {access codec/avcodec demux/avformat hw/vdpau packetizer video_chroma} {
+                foreach cf [glob -nocomplain ${worksrcpath}/modules/${module}/*.c] {
+                    reinplace "s|${dir}/|${dir}-VLC/|g" ${cf}
+                }
+                foreach cf [glob -nocomplain ${worksrcpath}/modules/${module}/*.h] {
+                    reinplace "s|${dir}/|${dir}-VLC/|g" ${cf}
+                }
+            }
+            reinplace "s|${dir}/|${dir}-VLC/|g" ${worksrcpath}/configure.ac ${worksrcpath}/configure
+        }
+        reinplace "s|postproc/|postproc-VLC/|g" ${worksrcpath}/configure.ac ${worksrcpath}/configure
+        reinplace "s|libpostproc/|libpostproc-VLC/|g" ${worksrcpath}/modules/video_filter/postproc.c
+        reinplace "s|HAVE_LIBAVUTIL_AV|HAVE_LIBAVUTIL_VLC_AV|g" \
+                ${worksrcpath}/modules/codec/avcodec/avcommon.h \
+                ${worksrcpath}/modules/codec/avcodec/avcommon_compat.h
+        reinplace "s|HAVE_LIBAVFORMAT_AV|HAVE_LIBAVFORMAT_VLC_AV|g" \
+                ${worksrcpath}/modules/codec/avcodec/avcommon.h \
+                ${worksrcpath}/modules/codec/avcodec/avcommon_compat.h
+        reinplace "s|HAVE_LIBAVCODEC_AV|HAVE_LIBAVCODEC_VLC_AV|g" \
+                ${worksrcpath}/modules/codec/avcodec/avcommon.h \
+                ${worksrcpath}/modules/codec/avcodec/avcommon_compat.h
+        reinplace "s|HAVE_LIBSWSCALE_SW|HAVE_LIBSWSCALE_VLC_SW|g" \
+                ${worksrcpath}/modules/codec/avcodec/avcommon.h \
+                ${worksrcpath}/modules/codec/avcodec/avcommon_compat.h
+    }
```

Comment:
Ugh, that's one giant hack. Can you not prepend the include path and library path of the private copy of ffmpeg to the include and library search paths for the build of VLC instead? That would be much cleaner and avoid patching upstream source.